pkg/yurthub/certificate/pki.go

```diff
@@ -83,3 +101,43 @@ func GenCertPoolUseCA(caFile string) (*x509.CertPool, error) {
 	certPool.AppendCertsFromPEM(caData)
 	return certPool, nil
 }
+
+// CheckValidityOfCertificate checks whether the certificate is valid or not
+func CheckValidityOfCertificate(cert *tls.Certificate) bool {
```

Comment:
how about modify `CheckValidityOfCertificate` to `hasServerAuthUsage`?

Reply:
The function name has been modified, please take review.